Generating Synthetic Code Review Comments from Bug Fix Histories

leusonmario opened the pull request:
**Goal:** Automatically generate synthetic code review comments by leveraging historical bug reports and their corresponding fixes.
**Input:** A list of previously reported bugs along with their associated fix implementations (commits).

marco-c commented:
CC @olewicki

leusonmario committed:
Generating comments from past bugs

suhaibmujahid commented:
The diff was difficult to review because it included 84 commits from the upstream. I cleaned up the branch and rebased it.

suhaibmujahid commented on scripts/comment_generation_bug_reports.py:
```diff
@@ -0,0 +1,705 @@
+            Subcategories include:
+                * None of the above
+                * Does not apply
+        - Keep It Focused: Limit your comments to the issues that could lead to problems identified by the Jira ticket and are directly related to the changes made in the Patch fixing the bug.
```
We could make it generic instead of specifying Jira here.

suhaibmujahid committed:
Fix incorrect database reference in docs (#5131)

Replaces the incorrect use of bugzilla.COMMITS_DB with repository.COMMITS_DB in the documentation example to ensure the correct database is downloaded.

leusonmario committed:
Adjusting indentation of prompts and technical terms
